### PR TITLE
feat(api-markdown-documenter): Allow filtering generated docs based on minimal release level

### PR DESCRIPTION
**Pull request by Josmithr**
This PR adds the `minimalReleaseLevel` configuration option, which allows consumers to specify a release tag to serve as the "minimal" level of API items to be processed when generating the API docs suite.

E.g. A user could specify `public` to only generate docs for the public API surface, or they could specify `beta` to include the API surface containing beta and public members (but not alpha or internal).

The motivation here is to enable hiding `@alpha` API members from the public (fluidframework.com) documentation, as those APIs are (currently) designated for pre-approved Microsoft-internal use only. Our internal API docs site can continue to show alpha docs.

This PR also updates a couple of the end-to-end test configurations to leverage this new functionality - you can see the resulting changes in the test snapshots.

AB#6469

### CODE REVIEW

File: tools/api-markdown-documenter/src/api-item-transforms/ApiItemTransformUtilities.ts [hunk 1]
```diff
@@ -459,3 +459,35 @@ function doesItemGenerateHierarchy(
...
+ * @param config - See {@link ApiItemTransformationConfiguration}.
+ */
+export function shouldItemBeIncluded(
+	apiItem: ApiItem,
+	config: Required<ApiItemTransformationConfiguration>,
+): boolean {
+	const releaseTag = getReleaseTag(apiItem);
+	if (releaseTag === undefined || releaseTag === ReleaseTag.None) {
+		// If the item does not have a release tag, then it inherits the release scope of its ancestry.
+		// If no release
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/ApiItemTransformUtilities.ts:479]:
```suggestion
```

**Review by zhenmichael — APPROVED** [review]
Overall LGTM!

File: tools/api-markdown-documenter/src/api-item-transforms/ApiItemTransformUtilities.ts [hunk 2]
```diff
@@ -459,3 +459,34 @@ function doesItemGenerateHierarchy(
...
+	const releaseTag = getReleaseTag(apiItem);
+	if (releaseTag === undefined || releaseTag === ReleaseTag.None) {
+		// If the item does not have a release tag, then it inherits the release scope of its ancestry.
+		const parent = getFilteredParent(apiItem);
+		if (parent === undefined) {
+			// If we encounter an item with no release tag in its ancestry, we can't make a determination as to whether
+			// or not it is intended to be included in the generated documentation suite.
+			// To be safe, log a warning but return true.
+			config.logger.warning("Encountered an API item with no release tag in ancestry.");
+			return true;
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/ApiItemTransformUtilities.ts:485]: Yeah, this is theoretically possible. Nested namespaces are one example, or the member of an interface / class under a namespace - if only the root namespace has a release tag, everything under it should inherit.

[the remark this answers is not in the document]
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/ApiItemTransformUtilities.ts:485]: I'll add this as an example in the comment to add some clarity.

**Pull request merged**